ironfish/src/rpc/routes/chain/getAssetInfo.ts

```diff
@@ -0,0 +1,73 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/. */
+ import { ASSET_IDENTIFIER_LENGTH } from '@ironfish/rust-nodejs'
+import * as yup from 'yup'
+import { Assert } from '../../../assert'
+ import { BlockHeader } from '../../../primitives'
+ import { BufferUtils } from '../../../utils'
+ import { ValidationError } from '../../adapters'
+ import { ApiNamespace, router } from '../router'
+ 
+ export type GetAssetInfoRequest = {
+   assetIdentifier: string
+ }
+ 
+ export type GetAssetInfoResponse = {
+    createdTransactionHash: string
+    identifier: string
+    metadata: string
+    name: string
+    nonce: number
+    owner: string
+    supply: number
+ }
+ 
+ export const GetAssetInfoRequestSchema: yup.ObjectSchema<GetAssetInfoRequest> = yup
+   .object()
+   .shape({
+      assetIdentifier: yup.string(),
+   })
+   .defined()
+ 
+ export const GetAssetInfoResponse: yup.ObjectSchema<GetAssetInfoResponse> = yup
+       .object({
+        createdTransactionHash: yup.string().defined(),
+        identifier: yup.string().defined(),
+        metadata: yup.string().defined(),
+        name: yup.string().defined(),
+        nonce: yup.number().defined(),
+        owner: yup.string().defined(),
+        supply: yup.number().defined(),
+   })
+   .defined()
+ 
+ router.register<typeof GetAssetInfoRequestSchema, GetAssetInfoResponse>(
+   `${ApiNamespace.chain}/getBlockInfo`,
+   GetAssetInfoRequestSchema,
+   async (request, node): Promise<void> => {
+     let header: BlockHeader | null = null
+     let error = ''
+ 
+     if (request.data.assetIdentifier.length != ASSET_IDENTIFIER_LENGTH) {
+        throw new ValidationError("Invalid asset identifier.")
+     }
+
+     let assetIdentifier = Buffer.from(request.data.assetIdentifier, 'hex')
+ 
+     const asset = await node.chain.assets.get(assetIdentifier)  
+ 
+     Assert.isNotUndefined(asset, 'no asset is found')
+
+     request.end({
+        createdTransactionHash: BufferUtils.toHuman(asset.createdTransactionHash),
+        identifier: BufferUtils.toHuman(asset.identifier),
+        metadata: BufferUtils.toHuman(asset.metadata),
```

Comment:
It's a field for users to put any extra info they want about an asset - there's no enforcement as to what they put in here besides a max length. I expect it to be primarily UTF8 strings

Reply:
Got it, that makes sense. We should hex encode this then and toHuman() this on the front end then or else devs cant get ths machine readable data here since toHuman() is lossy.